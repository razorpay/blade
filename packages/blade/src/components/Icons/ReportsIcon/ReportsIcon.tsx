import { Svg, Path } from '../_Svg';
import type { IconComponent } from '..';
import useIconProps from '../useIconProps';

const ReportsIcon: IconComponent = ({ size, color, ...styledProps }) => {
  const { height, width, iconColor } = useIconProps({ size, color });

  return (
    <Svg {...styledProps} width={width} height={height} viewBox="0 0 24 24" fill="none">
      <Path
        d="M4 6H2V20C2 21.1 2.9 22 4 22H18V20H4V6ZM20 2H8C6.9 2 6 2.9 6 4V16C6 17.1 6.9 18 8 18H20C21.1 18 22 17.1 22 16V4C22 2.9 21.1 2 20 2ZM19 11H9V9H19V11ZM15 15H9V13H15V15ZM19 7H9V5H19V7Z"
        fill={iconColor}
      />
    </Svg>
  );
};

export default ReportsIcon;
