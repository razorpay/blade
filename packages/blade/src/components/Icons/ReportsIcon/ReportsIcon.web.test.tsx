import ReportsIcon from './';
import renderWithTheme from '~src/_helpers/testing/renderWithTheme.web';

describe('<ReportsIcon />', () => {
  it('should render ReportsIcon', () => {
    const { container } = renderWithTheme(
      <ReportsIcon color="feedback.icon.neutral.lowContrast" size="large" />,
    );
    expect(container).toMatchSnapshot();
  });
});
