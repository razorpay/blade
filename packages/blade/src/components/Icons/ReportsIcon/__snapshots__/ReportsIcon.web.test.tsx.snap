// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportsIcon /> should render ReportsIcon 1`] = `
<div>
  <svg
    aria-hidden="true"
    data-blade-component="icon"
    fill="none"
    height="20px"
    viewBox="0 0 24 24"
    width="20px"
  >
    <path
      d="M4 6H2V20C2 21.1 2.9 22 4 22H18V20H4V6ZM20 2H8C6.9 2 6 2.9 6 4V16C6 17.1 6.9 18 8 18H20C21.1 18 22 17.1 22 16V4C22 2.9 21.1 2 20 2ZM19 11H9V9H19V11ZM15 15H9V13H15V15ZM19 7H9V5H19V7Z"
      fill="hsla(216, 27%, 36%, 1)"
    />
  </svg>
</div>
`;
